### ISSUE TITLE
[0.29] Packager server terminates with debug on, on Node 6.3.0

### ISSUE THREAD

**Issue by fungilation** [facebook/react-native#8668]
In my app, since updating to RN 0.29, I'm hitting the following redbox error:

```
Runtime is not ready for debugging. Make sure Packager server is running.
```

This happens consistently once I start testing my app in iOS 9.3 simulator. I reload, then same above redbox shows, and the Packager server then actually quits (also consistently). Error in Terminal before it quits:

```
 ERROR  out of range index
RangeError: out of range index
    at RangeError (native)
    at fastCopy (/Users/garyfung/Documents/code/flow_search/node_modules/ws/lib/Receiver.js:386:24)
    at Receiver.add (/Users/garyfung/Documents/code/flow_search/node_modules/ws/lib/Receiver.js:86:3)
    at Socket.realHandler (/Users/garyfung/Documents/code/flow_search/node_modules/ws/lib/WebSocket.js:800:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
    at Socket.Readable.push (_stream_readable.js:135:10)
    at TCP.onread (net.js:542:20)

See http://facebook.github.io/react-native/docs/troubleshooting.html
for common problems and solutions.
~
Process terminated. Press <enter> to close the window
```
Same error, screenshotted:
<img width="882" alt="screenshot 2016-07-08 23 53 17" src="https://cloud.githubusercontent.com/assets/3803466/16706588/44958c1e-4567-11e6-8ffe-f30aed9ae495.png">

I have not gotten any consistent quitting like this in the Packager server, before RN 0.29. Before updating today, I was on RN 0.25.

**Comment by fungilation** [comment]
Also, I see that if I turn off JS remote debugging, my CPU would get pegged. I turn on performance monitor and I see 0 fps on the JS thread. Seeing as my app was running fine prior to updating RN from 2.5 to 2.9, and that my CPU doesn't get pegged *unless* I turn off JS debugging, I don't think it's my app misbehaving.
<img width="487" alt="screenshot 2016-07-09 00 04 48" src="https://cloud.githubusercontent.com/assets/3803466/16706639/c8f52cf2-4568-11e6-824b-ae1589546ae1.png">

**Comment by sepehrooo** [comment]
I have this problem too.

**Comment by fungilation** [comment]
Exact same way, with pegged CPU when you turn off debugging, or only Package server consistently quitting?

**Comment by fab1an** [comment]
Try running `./node_modules/react-native/packager/packager.sh start --reset-cache`. Does that solve the issue?

**Comment by fungilation** [comment]
there are signs of life. I'll do more testing tomorrow

**Comment by ferndopolis** [comment]
i am also hitting a the same error recently running RN 0.24.  i did update chrome to latest v51.

**Comment by fungilation** [comment]
Chrome has nothing to do with this as it's caused by package server bugging out. I'm running Chrome 51 for a long while before I upgraded RN from 0.25 to 0.29. I doubt what you are seeing is the same bug if you just get a ```Runtime is not ready for debugging``` without the package server quitting like I'm seeing. I didn't get this on RN <= 0.25.

**Comment by JeandeCampredon** [comment]
We experimented the very same issue with different versions of node.
But node versions from 6.0.0 to 6.2.2 look safe
Hope it could help !

**Comment by jhen0409** [comment]
I've the same problem for node 6.3.

@JeandeCampredon thanks, it's work well when I back to 6.2 temporarily, very helpful. 👍

**Comment by ferndopolis** [comment]
@jhen0409 thanks that solve it, i did update my node to 6.3.0 last Friday.

**Comment by helmus** [comment]
Also having this with node 6.3 https://github.com/facebook/react-native/issues/8668

**Comment by fungilation** [comment]
Updated title to better reflect what needs fixing, to depend on Node < 6.3. I've moved to node4-lts/4.4.7 and I confirm package server stopped crashing (so far)

**Comment by ide** [comment]
My guess is this has to do with the ws dependency, which was upgraded recently with 0.30-rc.

**Comment by michaelAndersonCampingWorld** [comment]
I downgraded to 5.12, and it completely stopped crashing for about 3 days, but now I'm having a frustrating streak of runtime crashing. So, I'm not 100% convinced it was a problem with just node 6.3.

**Comment by ide** [comment]
Try 0.30 which was recently released.

**Comment by PaulBGD** [comment]
So for older RN versions (0.19) the fix is to use an older node version?

**Comment by pavlelekic** [comment]
I'm getting this all the time, it's so annoying. I'm using RN 0.29.

**Comment by ide** [comment]
Looking at the data points, you should either use Node 6.2 or RN 0.30 (make sure to delete all your node_modules and reinstall with RN 0.30).

**Comment by PaulBGD** [comment]
For anyone who wants an easy way to use Node 6.2 just for your React Native project, check out https://github.com/creationix/nvm

**Comment by marcellosachs** [comment]
This was also fixed for me once I switched to node 6.2.2 (using RN 0.29.2)

**Comment by ide** [comment]
Closing this since it should have been resolved with the `ws` dependency upgrade that went out awhile back with 0.30.
